lib/ae_mdw/db/mutations/key_block_statistics_mutation.ex

```diff
@@ -0,0 +1,49 @@
+defmodule AeMdw.Db.KeyBlockStatsMutation do
+  @moduledoc """
+  Creates a statistic records for key blocks.
+  """
+  alias AeMdw.Blocks
+  alias AeMdw.Db.Model
+  alias AeMdw.Db.Mutation
+  alias AeMdw.Db.State
+  alias AeMdw.Db.Sync.Stats
+
+  require Model
+
+  @derive Mutation
+  defstruct [:time, :difficulty]
+
+  @opaque t() :: %__MODULE__{
+            time: Blocks.time(),
+            difficulty: integer()
+          }
+
+  @spec new(Blocks.time(), integer()) :: t()
+  def new(time, difficulty), do: %__MODULE__{time: time, difficulty: difficulty}
+
+  @spec execute(t(), State.t()) :: State.t()
+  def execute(%__MODULE__{time: time, difficulty: difficulty}, state) do
+    time
+    |> Stats.time_intervals()
+    |> Enum.reduce(state, fn {interval_by, interval_start}, state ->
+      index = {:blocks_difficulty, interval_by, interval_start}
+
+      average_statistic(state, index, difficulty)
+    end)
+  end
+
+  defp average_statistic(state, index, new_difficulty) do
+    State.update(
+      state,
+      Model.Statistic,
+      index,
+      fn
+        Model.statistic(count: nil) = statistic ->
+          Model.statistic(statistic, count: new_difficulty)
+
+        Model.statistic(count: old_difficulty) = statistic ->
+          Model.statistic(statistic, count: round((old_difficulty + new_difficulty) / 2))
```

Comment:
This is not the average

Reply:
Yeah, the problem was that we can't write floats to the statistic table, I tried changing the type to accept pos_integers and floats, but it didn't work (It was still giving a serialization error)

Reply:
We already store the amount of blocks, we can simply accumulate the sum of difficulties and when we render then we divide them by the number of blocks

Reply:
That would be good, since it would also remove the separate mutation. Where are they stored?

Reply:
https://github.com/aeternity/ae_mdw/blob/master/lib/ae_mdw/db/sync/stats.ex#L101